ubuntu22.04/nvidia-driver

```diff
@@ -195,6 +195,36 @@ _assert_nvswitch_system() {
     return 0
 }
 
+_assert_nvlink5_system() (
+    for dir in /sys/class/infiniband/*/device; do
+        # Define the path to the VPD file
+        vpd_file="$dir/vpd"
+
+        # Check if the VPD file exists
+        if [ -f "$vpd_file" ]; then
+            # Search for 'SW_MNG' in the VPD file
+            if grep -q "SW_MNG" "$vpd_file"; then
+                echo "Detected NVLink5+ system"
+                return 0
+            fi
+        fi
+    done
+    return 1
+)
+
+_ensure_nvlink5_prerequisites() (
+    echo "Ensure that the ib_umad kernel module is loaded"
+    required_module_name="ib_umad"
```

Comment:
Do we not need to check for the mofed module too?

Reply:
Sure, we could. What I have observed was that ib_umad was only loaded after MOFED was installed and the node rebooted

Reply:
So I figured the mlx5_core module check was redundant

Reply:
If ib_umad depends on mlx5_core, then I guess only checking for ib_umad would suffice.

Reply:
Checking for both modules now. Just for good measure